x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/action/PutQueryRulesetAction.java

```diff
@@ -60,18 +61,24 @@ public Request(String rulesetId, BytesReference content, XContentType contentTyp
 
         @Override
         public ActionRequestValidationException validate() {
-            ActionRequestValidationException validationException = null;
+            AtomicReference<ActionRequestValidationException> validationException = new AtomicReference<>();
```

Comment:
I changed it because I'm modifying it in a lambda expression and wanted to make it effectively final.

Reply:
Ah, I see now. We could change that to use a standard for loop and remove the need for a lambda, thus removing the need for `AtomicReference` as well. I think that makes sense since `AtomicReference` will (unnecessarily, in this case) use a lock internally, WDYT?

Reply:
Yeah, that's fair.